### ISSUE TITLE
WSL2: Long UTF-8 filenames terminate entire process

### ISSUE THREAD

**Issue by HertzDevil** [microsoft/WSL#4219]
* Your Windows build number: `Microsoft Windows [Version 10.0.18917.1000]`

* What you're doing and what's happening:
  * Long filenames are enabled by setting `HKLM\SYSTEM\CurrentControlSet\Control\FileSystem\LongPathsEnabled` to `1`
  * Create a file in File Explorer with name `××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××××` (U+00D7 multiplication sign repeated 136 times)
  * Launch WSL2 in the same directory
  * Run `ls`

* What's wrong / what should be happening instead:
  * All WSL2 instances terminate. It doesn't matter which distro does the `ls`, *all* WSL2 distros terminate.
  * WSL legacy is able to display the file in `ls`. Not sure about WSL1.

**Comment by benhillis** [comment]
This sounds like a kernel panic caused by the 9p client in the Linux kernel.  Is there anything interesting in the Hyper-V worker process log in eventvwr?

**Comment by HertzDevil** [comment]
One critical level event:

```
'Virtual Machine' has encountered a fatal error.  The guest operating system reported that it failed with the following error codes: ErrorCode0: 0x0, ErrorCode1: 0x0, ErrorCode2: 0x0, ErrorCode3: 0x0, ErrorCode4: 0x0.  If the problem persists, contact Product Support for the guest operating system.  (Virtual machine ID 75503F4D-6CF2-5D75-8880-9D3574BCDE64)

Guest message:
<6>[    1.138623] libceph: loaded (mon/osd proto 15/24)
<6>[    1.138627] hv_vmbus: registering driver hv_sock
<6>[    1.138850] NET: Registered protocol family 40
<6>[    1.139941] registered taskstats version 1
<5>[    1.139952] Loading compiled-in X.509 certificates
<6>[    1.143697] Btrfs loaded, crc32c=crc32c-generic
<6>[    1.145753] rtc_cmos 00:00: setting system clock to 2019-06-22 18:05:55 UTC (1561226755)
<4>[    1.145790] Unstable clock detected, switching default tracing clock to "global"
<4>[    1.145790] If you want to keep using the local clock, then add:
<4>[    1.145790]   "trace_clock=local"
<4>[    1.145790] on the kernel command line
<6>[    1.188827] Freeing unused kernel image memory: 1348K
<6>[    1.229035] Write protecting the kernel read-only data: 18432k
<6>[    1.230438] Freeing unused kernel image memory: 2008K
<6>[    1.231120] Freeing unused kernel image memory: 1836K
<6>[    1.231462] Run /init as init process
<6>[    1.245190] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
<5>[    1.391077] scsi 0:0:0:0: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
<5>[    1.393256] sd 0:0:0:0: Attached scsi generic sg0 type 0
<5>[    1.403031] sd 0:0:0:0: [sda] 536870912 512-byte logical blocks: (275 GB/256 GiB)
<5>[    1.403036] sd 0:0:0:0: [sda] 4096-byte physical blocks
<5>[    1.404979] sd 0:0:0:0: [sda] Write Protect is off
<7>[    1.404981] sd 0:0:0:0: [sda] Mode Sense: 0f 00 00 00
<5>[    1.411261] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
<5>[    1.469546] sd 0:0:0:0: [sda] Attached SCSI disk
<6>[    2.019003] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
<6>[    2.571060] EXT4-fs (sda): 2 orphan inodes deleted
<6>[    2.571061] EXT4-fs (sda): recovery complete
<6>[    2.774771] EXT4-fs (sda): mounted filesystem with ordered data mode. Opts: discard,errors=remount-ro,data=ordered
<3>[    3.328605] FS-Cache: Duplicate cookie detected
<3>[    3.328609] FS-Cache: O-cookie c=000000005f741d3a [p=0000000056c02509 fl=222 nc=0 na=1]
<3>[    3.328610] FS-Cache: O-cookie d=00000000cec4f4cc n=00000000280fb814
<3>[    3.328611] FS-Cache: O-key=[10] '34323934393337353632'
<3>[    3.328619] FS-Cache: N-cookie c=000000004bc98b54 [p=0000000056c02509 fl=2 nc=0 na=1]
<3>[    3.328620] FS-Cache: N-cookie d=00000000cec4f4cc n=0000000005a2e23e
<3>[    3.328620] FS-Cache: N-key=[10] '34323934393337353632'
<3>[    3.345958] FS-Cache: Duplicate cookie detected
<3>[    3.345960] FS-Cache: O-cookie c=000000005040370b [p=0000000056c02509 fl=222 nc=0 na=1]
<3>[    3.345961] FS-Cache: O-cookie d=00000000cec4f4cc n=00000000729d3ce5
<3>[    3.345962] FS-Cache: O-key=[10] '34323934393337353634'
<3>[    3.345966] FS-Cache: N-cookie c=00000000e0f06e0c [p=0000000056c02509 fl=2 nc=0 na=1]
<3>[    3.345967] FS-Cache: N-cookie d=00000000cec4f4cc n=00000000344abc73
<3>[    3.345967] FS-Cache: N-key=[10] '34323934393337353634'
<6>[   49.900364] hv_balloon: Max. dynamic memory size: 7082 MB
<6>[  303.181863] traps: light-locker[1172] trap int3 ip:7f2f94ae3c75 sp:7ffd4351dec0 error:0 in libglib-2.0.so.0.5800.3[7f2f94aab000+7e000]
<0>[ 4066.319860] Kernel panic - not syncing: stack-protector: Kernel stack is corrupted in: v9fs_dir_readdir_dotl+0x1d1/0x1e0
<0>[ 4066.319860] 
<1>[ 4066.320954] CPU: 2 PID: 6843 Comm: bash Not tainted 4.19.43-microsoft-standard #1
<1>[ 4066.320959] Call Trace:
<1>[ 4066.322799]  dump_stack+0x5c/0x7b
<1>[ 4066.323428]  panic+0xe4/0x232
<1>[ 4066.323758]  ? v9fs_dir_readdir_dotl+0x1d1/0x1e0
<1>[ 4066.323765]  __stack_chk_fail+0x15/0x20
<1>[ 4066.323768]  v9fs_dir_readdir_dotl+0x1d1/0x1e0
<1>[ 4066.324102]  ? v9fs_dir_readdir+0x200/0x200
<1>[ 4066.324667]  ? iterate_dir+0x95/0x180
<1>[ 4066.324675]  ? ksys_getdents64+0x9d/0x120
<1>[ 4066.324679]  ? iterate_dir+0x180/0x180
<1>[ 4066.324683]  ? __x64_sys_getdents64+0x16/0x20
<1>[ 4066.324687]  ? __x64_sys_getdents64+0x16/0x20
<1>[ 4066.324729]  ? do_syscall_64+0x55/0x110
<1>[ 4066.324930]  ? entry_SYSCALL_64_after_hwframe+0x44/0xa9
<0>[ 4066.325862] Kernel Offset: disabled
```

Followed by an information level event:

```
'Virtual Machine' was shut down for a reset initiated by the guest operating system. (Virtual machine ID 75503F4D-6CF2-5D75-8880-9D3574BCDE64)
```

**Comment by benhillis** [comment]
Perfect exactly what I was looking for. This is either our server violating the protocol, or a bug in the Linux 9p client. @SvenGroot, can you take a look?

**Comment by therealkenc** [comment]
We have #3763 on the books too fwiw.

**Comment by sunilmut** [comment]
This should be fixed in upstream Linux kernel with [ef5305f1f72 et. al.](https://github.com/torvalds/linux/commit/ef5305f1f72eb1cfcda25c382bb0368509c0385b#diff-d9804c9e82370eeb6c891ee6f8f9eae4). I will see what can be done to get this in the WSL2 Linux kernel.

**Comment by therealkenc** [comment]
Weird state of affairs with WSL2,, because for any _qualified_ bug like this, there is technically a work-around-available. Work around being: "fix it". Strange times.
